### ISSUE TITLE
ChangeDetectorRef detached with detectChanges in a effect: UI doesn't update

### ISSUE THREAD

**Issue by cesco69** [angular/angular#50790]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

I have a component with detached `ChangeDetectorRef`. Into the signal `effect()` there is a `detectChanges()`

Why the component update the UI only if into the `effect()` there is a `counter()` call?

```ts
import 'zone.js/dist/zone';
import { Component, ChangeDetectorRef, signal, effect } from '@angular/core';
import { CommonModule } from '@angular/common';
import { bootstrapApplication } from '@angular/platform-browser';

@Component({
  selector: 'my-app',
  standalone: true,
  imports: [CommonModule],
  template: `
   {{ counter() }}
   <button (click)="incement()">Increment</button>
  `,
})
export class App {
  counter = signal(0);

  constructor(private cdr: ChangeDetectorRef) {
    this.cdr.detach();
    effect(() => {
      this.counter(); // remove this line and component don't update the ui
      this.cdr.detectChanges();
    });
  }

  incement() {
    this.counter.update((value) => value + 1);
  }
}

bootstrapApplication(App);
```

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-92tchj?file=src%2Fmain.ts

### Please provide the exception or error you saw

```true
No exception
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
_                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 16.1.0
Node: 18.12.1
Package Manager: pnpm 8.5.1
OS: win32 x64

Angular: 16.1.1
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, material-date-fns-adapter, platform-browser
... platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1601.0 (cli-only)
@angular-devkit/build-angular   16.1.0
@angular-devkit/core            16.1.0 (cli-only)
@angular-devkit/schematics      16.1.0 (cli-only)
@angular/cli                    16.1.0
@schematics/angular             16.1.0 (cli-only)
rxjs                            7.8.1
typescript                      5.1.3
```

**Comment by cesco69** [comment]
Side question:

Why Signals also need a detectChanges()? Angular always call the effect on signal update

**Comment by JeanMeche** [comment]
About the side question: Signals are still hooked onto the old change detection system. So if you call `detach()`, you won't have the automatic change detection.
